### PR TITLE
Fix test_conftest when run via pytest-randomly

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

File: testing/test_conftest.py [hunk 1]
```diff
@@ -15,13 +16,24 @@
 
 @pytest.fixture(scope="module", params=["global", "inpackage"])
 def basedir(request, tmpdir_factory):
+    # Do not write bytecode to make tests not fail when run randomly
+    # (ConftestImportFailure with TestConftestValueAccessGlobal).
+    orig_env = os.environ.get("PYTHONDONTWRITEBYTECODE")
```
Comment (nicoddemus) [inline testing/test_conftest.py]: I suggest to create a `MonkeyPatch()` instance manually and call `undo` at the end

Comment (blueyed) [reply testing/test_conftest.py]: Turned out it was related to sys.modules instead - used SysModulesSnapshot there now.

**Comment by blueyed** [comment]
I wonder if we should have a strict CI job, which then would use this:
```diff
diff --git c/testing/conftest.py i/testing/conftest.py
new file mode 100644
index 00000000..b4856828
--- /dev/null
+++ i/testing/conftest.py
@@ -0,0 +1,13 @@
+import sys
+
+import pytest
+
+
+@pytest.fixture(autouse=True)
+def _check_syspath():
+    """Ensure that tests do not mess with sys.path."""
+    oldpath = sys.path[:]
+    yield
+    assert (
+        oldpath == sys.path
+    ), "sys.path was changed in test, consider using _sys_snapshot"
```

**Comment by nicoddemus** [comment]
> I wonder if we should have a strict CI job

Excellent idea, but do you mean a separate job? I think that fixture should be included in every normal run, it is cheap and catches mistakes early. 👍

**Comment by blueyed** [comment]
Wrapping every test is not that cheap, is it? :)
But if it's OK we could just have it by default already (i.e. not only on CI).

**Comment by nicoddemus** [comment]
> Wrapping every test is not that cheap, is it? :)
But if it's OK we could just have it by default already (i.e. not only on CI).

I would say extremely cheap if compared to all the other things that are happening behind the scenes; just to put things into perspective, consider how much works is done just to call a hook... having an extra fixture which will cost a few ms to the end of the test run is definitely worth it if we are avoiding mistakes in the process. 😉

**Comment by blueyed** [comment]
I think the check should be added separately, and IMHO it is enough to run this from time to time manually (like I did here).

**Pull request merged**